spec/routes/api/committee_spec.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require_relative "spec_helper"
+
+RSpec.describe Clover, "committee infrastructure" do
+  let(:user) { create_account }
+  let(:project) { user.create_project_with_default_policy("project-1") }
+
+  before do
+    login_api(user.email)
+  end
+
+  it "serializes unexpected nested errors that cannot be converted" do
+    ex = Class.new(Committee::BadRequest) do
+      def self.name
+        "CommitteeTestNestedException"
+      end
+
+      def original_error
+        Exception.new("unaddressed error value")
+      end
+    end.new("testing conversion of unenumerated nested exceptions")
+
+    allow_any_instance_of(Committee::SchemaValidator::OpenAPI3).to receive(:request_validate) do
+      raise ex
+    end
+
+    expect {
+      post "/project/#{UBID.generate(UBID::TYPE_PROJECT)}/location/#{TEST_LOCATION}/vm/test-vm"
+    }.to raise_error ex
+  end
+
+  it "rejects paths that cannot be found in the schema" do
+    post "/not-a-prefix/"
+    expect(JSON.parse(last_response.body).dig("error", "message")).to eq("Sorry, we couldn’t find the resource you’re looking for.")
+  end
+
+  it "fails when the request has unparsable json" do
+    expect {
+      post "/project/#{project.ubid}/location/#{TEST_LOCATION}/vm/test-vm", "this is not json"
+    }.to raise_error Committee::InvalidRequest, "Request body wasn't valid JSON."
+  end
+
+  it "fails when response has invalid structure" do
+    expect(Serializers::Vm).to receive(:serialize_internal).and_return({totally_not_correct: true})
+    expect {
+      post "/project/#{project.ubid}/location/#{TEST_LOCATION}/vm/test-vm", {
+        public_key: "ssh key",
+        unix_user: "ubi",
+        size: "standard-2",
+        boot_image: "ubuntu-jammy",
+        storage_size: "40"
+      }.to_json
+    }.to raise_error Committee::InvalidResponse, %r{#/components/schemas/Vm missing required parameters: .*}
+  end
+
+  it "fails when request is missing required keys" do
+    expect {
+      post "/project/#{project.ubid}/location/#{TEST_LOCATION}/vm/test-vm", {}.to_json
+    }.to raise_error Committee::InvalidRequest, "#/paths/~1project~1{project_id}~1location~1{location}~1vm~1{vm_name}/post/requestBody/content/application~1json/schema missing required parameters: public_key"
+  end
+
+  it "fails when request has extra unexpected keys" do
+    expect {
+      post "/project/#{project.ubid}/location/#{TEST_LOCATION}/vm/test-vm", {missing_stuff: true}.to_json
+    }.to raise_error Committee::InvalidRequest, "#/paths/~1project~1{project_id}~1location~1{location}~1vm~1{vm_name}/post/requestBody/content/application~1json/schema does not define properties: missing_stuff"
+  end
+
+  it "reports if response body is not valid JSON" do
+    allow_any_instance_of(Committee::SchemaValidator::OpenAPI3).to receive(:response_validate) do
```

Comment:
Avoiding `allow_any_instance_of` here could probably be done by overriding `Serializers::Vm.serialize` to return an invalid JSON string (maybe that triggers a different issue because the response header isn't set correctly).

Reply:
well per minor abuse of the term, where serialize is more like "idiomatic downprojected representation of this rich object" (and leaves to_json to a final step) I don't think that'll work to get a invalid response output:

```ruby
    def self.serialize(object, options = {})
      return if object.nil?
      if object.respond_to?(:map) && !object.is_a?(Hash)
        object.map { |item| serialize_internal(item, options) }
      else
        serialize_internal(object, options)
      end
    end
```